README.md

```diff
@@ -97,6 +97,69 @@ Book::STATES['in_weird_state'] # 'in weird State'
 
 ```
 
+#### NxtSupport::HashTranslator
+
+`NxtSupport::HashTranslator` is a module that allows you to manipulate keys and values of original hash through tuple hash. 
+Tuple hash is a hash where `key` - represent's the key of original hash, and `value` - represents the key of result hash.
+Use `#translate_hash` method to get the result hash.
+
+```ruby
+class TestClass
+  include NxtSupport::HashTranslator
+end
+
+TestClass.translate_hash(firstname: 'John', firstname: :first_name)
+=> { 'first_name' => 'John' }
+```
+The `value` also could be a `Hash`. If this hash contain only 1 pair `key` would be used as a new `key`
+and `value` would be used as it is unless it responds to `#call`. In this cases `#call` would be called and actual value
+passed to the block called (if no, original value would be left). In case there is more than 1 pair first `key` would
+be used and last `value` would be used as well. 
+
+```ruby
+class TestClass
+  include NxtSupport::HashTranslator
+end
+# Multiple pair hash used as hash value
+hash = { firstname: 'John', phonenumber: '11-22-33-445' }
+tuple = {
+          firstname: :first_name,
+          phonenumber: {
+            phone_number: ->(number) { number.to_s.prepent('+49') },
```

Comment:
The lambda should be removed from this example, unless I don't understand it correctly.

Reply:
So it is actually do so - if you make a hash as value in tuples in this internal hash key would be a new key, but value, if it responds to `#call` would be used to process input hash actual value. I did not check why we need it, I just get the most wide implementation of `Util::HashTranslator` so `NxtSupport::HashTranslator` would not break anything in any of our projects.

Reply:
If understand this example correctly then `phone_number` should have `+49` prepended to it.  However, the return value shown at line `134` does not have `+49` prepended to it.

Reply:
That's how it worked - took the first key nested hash but the last value. Discussed this behaviour with @getand and decided to raise an error when you try to pass the nested hash with more than 1 key-value pair. It would be more clear and left no questions regarding behaviour.